File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        answer = []

        dictnaory = {}

        for num in range(len(nums)) :
            dictnaory[nums[num]] = num

        for num in range(len(nums)):
            helper = target - nums[num]
            if dictnaory.__contains__(helper) and dictnaory.get(helper) != num :
                answer.append(num)
                answer.append(dictnaory.get(helper))
                

                return answer

        return answer

File: Data Structures & Algorithms/binary-search/submission-2.java
class Solution {
    public int search(int[] nums, int target) {
        int n = nums.length;
        int start = 0;
        int end = n-1;

        while(start <= end){
            int m = (start + end)/2;
            System.out.print(m);
            if(nums[m] == target){
                System.out.print("herhe");
                return m;
            }else if(nums[m] < target){
                start = m+1;
            }else{
                end = m-1;
            }
        }
        return -1;
    }
}
